examples/http/main.go

```diff
@@ -0,0 +1,203 @@
+		fmt.Fprintf(os.Stderr, "It can run as either a server or client.\n\n")
+		fmt.Fprintf(os.Stderr, "Options:\n")
+		flag.PrintDefaults()
+		fmt.Fprintf(os.Stderr, "\nExamples:\n")
+		fmt.Fprintf(os.Stderr, "  Run as server:  %s server\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "  Run as client:  %s client\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "  Custom host/port: %s server https://0.0.0.0:8000\n", os.Args[0])
+		os.Exit(1)
+	}
+
+	if len(os.Args) < 2 {
+		fmt.Fprintf(os.Stderr, "Error: Must specify 'client' or 'server' as first argument\n\n")
+		flag.Usage()
+	}
+	mode := os.Args[1]
+
+	rawurl := "http://localhost:8000"
+	if len(os.Args) >= 3 {
+		rawurl = os.Args[2]
+	}
+	url, err := url.Parse(rawurl)
+	if err != nil {
+		log.Fatalf("Server failed: %v", err)
+	}
+
+	switch mode {
+	case "server":
+		runServer(url)
+	case "client":
+		runClient(url)
+	default:
+		fmt.Fprintf(os.Stderr, "Error: Invalid mode '%s'. Must be 'client' or 'server'\n\n", mode)
+		flag.Usage()
+	}
+}
+
+// GetTimeParams defines the parameters for the cityTime tool.
+type GetTimeParams struct {
+	City string `json:"city" jsonschema:"City to get time for (nyc, sf, or boston)"`
+}
+
+// getTime implements the tool that returns the current time for a given city.
+func getTime(ctx context.Context, ss *mcp.ServerSession, params *mcp.CallToolParamsFor[GetTimeParams]) (*mcp.CallToolResultFor[any], error) {
+	// Define time zones for each city
+	locations := map[string]string{
+		"nyc":    "America/New_York",
+		"sf":     "America/Los_Angeles",
+		"boston": "America/New_York",
+	}
+
+	city := params.Arguments.City
+	if city == "" {
+		city = "nyc" // Default to NYC
+	}
+
+	// Get the timezone.
+	tzName, ok := locations[city]
+	if !ok {
+		return nil, fmt.Errorf("unknown city: %s", city)
+	}
+
+	// Load the location.
+	loc, err := time.LoadLocation(tzName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load timezone: %w", err)
+	}
+
+	// Get current time in that location.
+	now := time.Now().In(loc)
+
+	// Format the response.
+	cityNames := map[string]string{
+		"nyc":    "New York City",
+		"sf":     "San Francisco",
+		"boston": "Boston",
+	}
+
+	response := fmt.Sprintf("The current time in %s is %s",
+		cityNames[city],
+		now.Format(time.RFC3339))
+
+	return &mcp.CallToolResultFor[any]{
+		Content: []mcp.Content{
+			&mcp.TextContent{Text: response},
+		},
+	}, nil
+}
+
+func runServer(url *url.URL) {
+	// Create an MCP server.
+	server := mcp.NewServer(&mcp.Implementation{
+		Name:    "time-server",
+		Version: "1.0.0",
+	}, nil)
+
+	// Add the cityTime tool.
+	mcp.AddTool(server, &mcp.Tool{
+		Name:        "cityTime",
+		Description: "Get the current time in NYC, San Francisco, or Boston",
+	}, getTime)
+
+	// Create the streamable HTTP handler.
+	handler := mcp.NewStreamableHTTPHandler(func(req *http.Request) *mcp.Server {
+		return server
+	}, nil)
+
+	handlerWithLogging := loggingHandler(handler)
+
+	laddr := fmt.Sprintf("%s:%s", url.Hostname(), url.Port())
```

Comment:
Allowing proto is deceptive: I can't create an HTTPS server if I write `https://example.com:8080`. 
I think you should go back to host and port flags. The host will essentially always be localhost, so that should definitely be a flag. If you want to have port be an arg, I'm OK with that, but I'm also fine with a flag and a default value.

Reply:
@jba take a look -- I fixed the rest of stuff, and I want to keep the proto, because without a way to set it to https:// for the client, I can't really test the remote side, if it's behind the load balancer.